src/com/google/enterprise/adaptor/CronScheduler.java

```diff
@@ -72,7 +72,7 @@ public void reschedule(Future<?> future, String pattern) {
     if (!(future instanceof CronFuture)) {
       throw new IllegalArgumentException("Provided future is not a CronFuture");
     }
-    CronFuture cronFuture = (CronFuture) future;
+    CronFuture<?> cronFuture = (CronFuture) future;
```

Comment:
The updated line adds "<?>".  

This addition seems to make less readable to me.  

What do you think?

Reply:
I think it's the language we have. I really dislike suppressing warnings. It's harmless in cases like this, but not everything that could get suppressed is harmless, and with the warnings suppressed we write code we didn't mean to (e.g., the List<Map.Entry> stuff in DocIdSenderTest).

Reply:
What's wrong with List<Map.Entry> in DocIdSenderTest?